Make plex detect subdirectories as different series

harshithmohan opened the pull request:
Fixes issue #4

harshithmohan commented:
The log is getting too big and being rotated frequently because `mediaList` does become empty, I need to see if I can fix that.

Cazzar commented:
Feel free to not log medialist

On Sun, 10 May 2020, 11:38 pm Harshith Mohan, <notifications@github.com>
wrote:

> The log is getting too big and being rotated frequently because mediaList
> does become empty, I need to see if I can fix that.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Cazzar/ShokoMetadata.bundle/pull/39#issuecomment-626329461>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHSXJW7DIEGQQT5E3EP3ZDRQ2U4XANCNFSM4M5GMVGQ>
> .
>

harshithmohan commented:
Yeah, did that. Doesn't help much anyway.

harshithmohan commented:
~~**DO NOT MERGE**
I'm trying to fix the double scanning problem~~
nvm, fixed it

harshithmohan commented:
Fixed the double-scanning. As far as re-scanning goes, group folders **will** get re-scanned everytime a library scan is run because we are calling the scan on them manually or beacuse they are relative paths. I don't know which of them is the problem but it exists in Absolute Series Scanner too, so that's for another day.

Also, I added a preference for including movies as Elemental Crisis suggested that would be more user-friendly than searching for and deleting a line.

harshithmohan commented:
At least an option, as some people do want all anime to be in the same library?
But I can revert it if you want, no problem.

Cazzar commented:
It's something I put in explicitly, and with the setup that I am looking at, it will be possible to scan movies and series from the same root folder.

harshithmohan commented:
Ok then, I'll revert that.

Cazzar commented on Contents/Resources/Series/Shoko Series Scanner.py:
```diff
@@ -95,12 +95,14 @@ def GetApiKey():
-        Log.debug('subdirs: %s', subdirs)
-        Log.debug('language: %s', language)
-        Log.info('root: %s', root)
+
```
Any reason for this added whitespace?

harshithmohan replied:
Nope, that's an accident.

Cazzar commented on Contents/Resources/Series/Shoko Series Scanner.py:
```diff
@@ -145,10 +147,50 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
+                Scan(path, sorted(subdir_files), mediaList, [], language=language, root=root) 
+                # relative path for dir or it will group multiple series into one as before and no empty subdirs array because they will be scanned afterwards.
+            
+            Log.info(''.ljust(100, '='))
```
isn't `'=' * 100` easier?

Note: I had to pull up a repl to confirm that is the intended function

Cazzar commented on Contents/Resources/Series/Shoko Series Scanner.py:
```diff
@@ -145,10 +147,50 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
+            
+            Log.info(''.ljust(100, '='))
+            Log.info('Completed subfolder scan: %s', full_path)
+            Log.info(''.ljust(100, '='))
```
as above

Cazzar commented on Contents/Resources/Series/Shoko Series Scanner.py:
```diff
@@ -145,10 +147,50 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
+                if grouping_dir in subdirs:
+                    subdirs.remove(grouping_dir)  #Prevent group folders from being called by Plex normal call to Scan()
+                Log.info("CALLING SCAN FOR FILES IN CURRENT FOLDER")
+                Scan(path, sorted(subdir_files), mediaList, [], language=language, root=root) 
```
do the parameters have to be explicit here? since they are in a defined order above, and you keep that same naming convention `root=root` seems redundant, and seems more like a local variable assignment more than a param

harshithmohan replied:
Again, copied code.. xD

Cazzar commented on Contents/Resources/Series/Shoko Series Scanner.py:
```diff
@@ -145,10 +147,50 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
+
+            reverse_path = list(reversed(path.split(os.sep)))
+            
+            Log.info(''.ljust(100, '='))
```
same as 191

Cazzar commented on Contents/Resources/Series/Shoko Series Scanner.py:
```diff
@@ -145,10 +147,50 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
+            
+            Log.info(''.ljust(100, '='))
+            Log.info('Started subfolder scan: %s', full_path)
+            Log.info(''.ljust(100, '='))
```
same as 191

Cazzar commented on Contents/DefaultPrefs.json:
```diff
@@ -31,7 +31,7 @@
     {
         "id": "SingleSeasonOrdering",
-        "label": "Use single season ordering (IMPORTANT: Make sure the setting in scanner is matches with this!)",
+        "label": "Use single season ordering (IMPORTANT: Make sure the setting in scanner matches with this!)",
```
Generally, for a merge request, you wouldn't want to add unrelated stuff, but I'll let this through :)

harshithmohan replied:
My brain didn't allow me to have a typo there. xD
Had to do it.

harshithmohan commented:
UPDATE: Been using this primarily on my whole library since 2 days. Everything's working and no problems found.

harshithmohan commented:
There's an issue with partial scan.
When plex starts a partial scan on a group folder, the subfolders aren't being considered as different series
EDIT: This is when adding multiple folders to collection folder. The partial scan which runs when adding an episode works properly.
EDIT 2: Seems unfixable, even if the root call is imitated with just the collection folder as a sub-directory, Plex doesn't want to accept they are different series.
Absolute Series Scanner too just skips those folders during the partial scan.
